lib/tapioca/commands/dsl.rb

```diff
@@ -363,7 +363,8 @@ def underscore(class_name)
 
       sig { params(constant: String).returns(String) }
       def rbi_filename_for(constant)
-        underscore(constant) + ".rbi"
+        # TODO: Figure out a better way than this.
+        underscore(constant).gsub("rubo_cop", "rubocop") + ".rbi"
```

Comment:
Apologies for setting you on the wrong path, but let's switch the name of the compiler to `Rubocop` instead.

Reply:
Sorry these are two separate things (I originally erroneously thought they were related):

- **`Tapioca::Commands::DSL#rbi_filename_for` controls the output file name for the generated RBIs**, which are based on the name of the constant they were generated for. This is hard to override for just constants with `RuboCop` in their name (and we don't control those constant names), and probably not worth doing (as @Morriar [suggested](https://github.com/Shopify/tapioca/pull/1046#discussion_r1008180704)).
- **`DslSpec#target_class_file` controls which compiler file is required based on the name of the spec file.** Fortunately, that module is included in the specs, so we can just override the method in the spec to say that we want to `require` `"tapioca/dsl/compilers/rubocop.rb"` instead of `"tapioca/dsl/compilers/rubo_cop.rb"`, which makes it trivial for Tapioca to **internally** match both the `RuboCop` constant name and `rubocop` file name conventions the RuboCop gem follows.

Reply:
Oh, I see what you mean. It's not related to the DSL compiler class. That's unfortunate for sure, but not sure if there's a better way of doing this other than treating rubocop as a special case

Reply:
Yeah, exactly, and if we were to allow for that (perhaps through a configurable inflector), it would probably be a change that doesn't belong in this PR.